src/toil/lib/aws/iam.py

```diff
@@ -0,0 +1,182 @@
+
+import logging
+import boto3
+from toil.lib.aws import zone_to_region
+from toil.provisioners.aws import get_best_aws_zone
+from functools import lru_cache
+from typing import Any, List, Dict, Set, cast
+
+from mypy_boto3_iam import IAMClient
+from mypy_boto3_iam.type_defs import GetRolePolicyResponseTypeDef
+from toil.lib.aws.session import AWSConnectionManager
+
+
+logger = logging.getLogger(__name__)
+
+_CLUSTER_LAUNCHING_PERMISSIONS = {"iam:CreateRole",
+                                  "iam:CreateInstanceProfile",
+                                  "iam:TagInstanceProfile",
+                                  "iam:DeleteRole",
+                                  "iam:DeleteRoleProfile",
+                                  "iam:ListAttatchedRolePolicies",
+                                  "iam:ListPolicies",
+                                  "iam:ListRoleTags",
+                                  "iam:PutRolePolicy",
+                                  "iam:RemoveRoleFromInstanceProfile",
+                                  "iam:TagRole"
+                                  }
+
+
+def check_policy_warnings(allowed_actions: Dict[str, List[str]] = {'*': []}, launching_perms: Set[str] = _CLUSTER_LAUNCHING_PERMISSIONS) -> bool:
```

Comment:
This function should have a better name.

If I see `y = check_policy_warnings(x)`, then `y` is probably a `bool`, but I'm going to assume that `x` somehow represents "policy warnings" that need to be checked. And that's not what's happening at all.

If you want to usually call it with only the one argument, you might want to call it something like `permission_grants_allow_launching_cluster`. Personally I think it might be even better to have the user import both the function and the permissions list, so the calls would look something like:

```
if not permission_grants_allow(grants, CLUSTER_LAUNCHING_PERMISSIONS):
    logger.warning('You may not have sufficient permissions!')
```

It's very clear what that code is doing, versus:

```
if not check_policy_warnings(allowed_actions):
    # Does the reader have any idea what the code now knows? If the sense of the if statement was backward, how quickly would you be able to tell?
```